mongodbatlas/resource_mongodbatlas_custom_db_role.go

```diff
@@ -101,7 +102,13 @@ func resourceMongoDBAtlasCustomDBRole() *schema.Resource {
 	}
 }
 
+var (
+	customRoleLock sync.Mutex
+)
+
 func resourceMongoDBAtlasCustomDBRoleCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	customRoleLock.Lock()
+	defer customRoleLock.Unlock()
```

Comment:
[q] Could you give me more details about why we hit this issue? Thanks

Reply:
Atlas API if you run each terraform resource apply sequentially in slow motion i.e. debugger it updates state correctly
If you run all in parallel API does not come back with consistent values Zuhair's example test 3 separate roles that fail if run all together

Reply:
@andreaangiolillo brief example here: https://drive.google.com/file/d/1aDfPwxjXm35qkEQgP135M6FxZAss5Qlv/view
also in body of jira ticket as well if you need: https://jira.mongodb.org/browse/INTMDB-448